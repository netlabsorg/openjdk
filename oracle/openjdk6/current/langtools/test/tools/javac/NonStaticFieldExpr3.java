/* @test  /nodynamiccopyright/
   @bug 4087127 4785453
   @author dps
   @summary class: instance access through types is not allowed

   @run shell NonStaticFieldExpr3.sh
*/

class NonStaticFieldExpr3 {
  public int x;
}

class Subclass extends NonStaticFieldExpr3 {
  int a = NonStaticFieldExpr3.x;      // SHOULD BE ERROR
}
