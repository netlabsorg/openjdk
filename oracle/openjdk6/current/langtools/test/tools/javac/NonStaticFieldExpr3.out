NonStaticFieldExpr3.java:14:30: compiler.err.non-static.cant.be.ref: (- compiler.misc.kindname.variable), x
1 error
